src/store/io.rs

```diff
@@ -150,15 +150,15 @@ fn submission_entry(
             buf_ptr,
             PAGE_SIZE as u32,
         )
-        .offset(page_id * PAGE_SIZE as u64)
+        .offset((store.data_page_offset + page_id) * PAGE_SIZE as u64)
```

Comment:
no, i don't think this is correct. because we also need to write meta pages.

the best approach would be to introduce an enum, something like

```rust
enum PageIndex {
  DataPage(u64),
  MetaBytePage(u64),
  MetaPage,
}
```

Reply:
I will merge this but will implement this in a follow up.

Reply:
you're totally right, perfect solution in #6